### PR TITLE
Rework check-time so it is more verbose, asserts its config always exists

### PR DESCRIPTION
**Pull request by cmaloney**

### CODE REVIEW

File: packages/check-time/extra/check-time.c [hunk 1]
```diff
@@ -12,29 +12,58 @@ int main() {
...
+    } else if (strcmp(enable_check_time, "true") == 0) {
+        // Fall out and check time is synchronized.
+    } else {
+        fprintf(stderr, "ENABLE_CHECK_TIME is something other than 'true' or 'false'. "
+            "internal consistency is broken. failing hard.\n");
+        return 1;
+    }
+
+
+    fprintf(stderr, "Checking whether time is synchronized using the kernel adjtimex API.")
```
Comment (sargun) [inline packages/check-time/extra/check-time.c]: \n intended at the end of this? and the next line?

File: packages/check-time/extra/check-time.c [hunk 2]
```diff
@@ -12,29 +12,58 @@ int main() {
...
-	error = tx.esterror - MAX_EST_ERROR_US;
+    if (rc == -1) {
+        perror("adjtimex");
+        fprintf(stderr, "adjtimex() returned -1 indicating error. Unable to determine clock sync. See above perror message for details.");
+        return 1;
+    }
+
+	if (rc == TIME_ERROR) {
+		fprintf(stderr, "Time is not synchronized / Marked as bad by the kernel.\n");
+        return 1;
```
Comment (sargun) [inline packages/check-time/extra/check-time.c]: Is this github being weird, is alignment weird here?

File: packages/check-time/extra/check-time.c [hunk 3]
```diff
@@ -12,29 +12,58 @@ int main() {
...
-	else
-		return 0;
-	return 1;
+	if (tx.status & STA_UNSYNC) {
+		fprintf(stderr, "Clock is out of sync / in unsync state. Must be synchronized for proper operation.\n");
+        return 1;
+    }
+
+    // All time sync check passed, clock is in sync.
+    fprintf(stderr, "Time is in sync!\n")
```
Comment (sargun) [inline packages/check-time/extra/check-time.c]: does this make sense to pump out via stderr, or stdout?

File: packages/check-time/extra/check-time.c [hunk 4]
```diff
@@ -12,29 +12,58 @@ int main() {
 	int rc;
 	long long int error;
 
-	if (getenv("IGNORE_CHECK_TIME")) {
-		fprintf(stderr, "check-time ignoring time, passing no matter what\n");
-		return 0;
-	}
+    const char *enable_check_time = getenv("ENABLE_CHECK_TIME")
```
Comment (sargun) [inline packages/check-time/extra/check-time.c]: semicolon?

File: packages/check-time/extra/check-time.c (hunk 1 above)
Comment (orsenthil) [inline packages/check-time/extra/check-time.c]: also, semicolon; at the end.

File: packages/check-time/extra/check-time.c [hunk 5]
```diff
@@ -12,29 +12,58 @@ int main() {
...
 
 	rc = adjtimex(&tx);
-	error = tx.esterror - MAX_EST_ERROR_US;
+    if (rc == -1) {
+        perror("adjtimex");
+        fprintf(stderr, "adjtimex() returned -1 indicating error. Unable to determine clock sync. See above perror message for details.");
+        return 1;
+    }
+
+	if (rc == TIME_ERROR) {
```
Comment (cmaloney) [inline packages/check-time/extra/check-time.c]: http://man7.org/linux/man-pages/man2/adjtimex.2.html
```
                   The symbolic name TIME_BAD is a synonym for TIME_ERROR,
                   provided for backward compatibility.
```

File: packages/check-time/extra/check-time.c (hunk 3 above)
Comment (orsenthil) [inline packages/check-time/extra/check-time.c]: semi-colon;

File: packages/check-time/extra/check-time.c (hunk 5 above)
Comment (sargun) [inline packages/check-time/extra/check-time.c]: TIME_ERROR didn't seem to be available on my machine. :-/?

File: packages/check-time/extra/check-time.c (hunk 2 above)
Comment (sargun) [inline packages/check-time/extra/check-time.c]: Yeah, but the if, and return aren't lined up.

**Pull request merged**